Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Beckn/Status.hs

```diff
@@ -44,23 +52,43 @@ data DStatusRes = StatusRes
   }
 
 handler ::
-  (CacheFlow m r, EsqDBFlow m r) =>
+  ( CacheFlow m r,
+    EsqDBFlow m r,
+    EsqDBFlow m r,
+    EncFlow m r,
+    HasFlowEnv m r '["nwAddress" ::: BaseUrl],
+    HasHttpClientOptions r c,
+    HasLongDurationRetryCfg r c,
+    CoreMetrics m
+  ) =>
   Id DM.Merchant ->
   DStatusReq ->
   m DStatusRes
 handler transporterId req = do
-  transporter <-
-    QM.findById transporterId
-      >>= fromMaybeM (MerchantNotFound transporterId.getId)
-  ride <- QRide.findById req.rideId >>= fromMaybeM (RideDoesNotExist req.rideId.getId)
-  booking <- QRB.findById ride.bookingId >>= fromMaybeM (BookingNotFound ride.bookingId.getId)
-  let transporterId' = booking.providerId
-  unless (transporterId' == transporterId) $ throwError AccessDenied
-
-  return $
-    StatusRes
-      { bookingId = booking.id,
-        bookingStatus = booking.status,
-        rideStatus = ride.status,
-        ..
-      }
+  transporter <- QM.findById transporterId >>= fromMaybeM (MerchantNotFound transporterId.getId)
+  booking <- QRB.findById req.bookingId >>= fromMaybeM (BookingNotFound req.bookingId.getId)
+  case booking.bookingType of
+    DRB.NormalBooking -> do
+      ride <- QRide.findByRBId booking.id >>= fromMaybeM (RideDoesNotExist booking.id.getId)
+      let transporterId' = booking.providerId
+      unless (transporterId' == transporterId) $ throwError AccessDenied
+      return $
+        StatusRes
+          { bookingId = booking.id,
+            bookingStatus = booking.status,
+            rideStatus = ride.status,
+            ..
+          }
+    DRB.SpecialZoneBooking -> do
+      Esq.runTransaction $ QBooking.updateStatus req.bookingId DRB.CANCELLED
+      merchant <- CQM.findById transporterId >>= fromMaybeM (MerchantNotFound transporterId.getId)
+      CallBAP.sendBookingCancelledUpdateToBAP booking merchant SRBCR.ByApplication
```

Comment:
Why status api handler call `on_update`? it already call `on_status`

Reply:
pushed new changes